src/ui/utils/utils.go

```diff
@@ -42,33 +42,52 @@ func ScanAllImages() error {
 		log.Errorf("Failed to list all repositories, error: %v", err)
 		return err
 	}
-	log.Infof("Rescanning all images.")
+	log.Infof("Scanning all images on Harbor.")
 
-	go func() {
-		var repoClient *registry.Repository
-		var err error
-		var tags []string
-		for _, r := range repos {
-			repoClient, err = NewRepositoryClientForUI(regURL, true, "harbor-ui", r.Name, "pull")
-			if err != nil {
-				log.Errorf("Failed to initialize client for repository: %s, error: %v, skip scanning", r.Name, err)
-				continue
-			}
-			tags, err = repoClient.ListTag()
-			if err != nil {
-				log.Errorf("Failed to get tags for repository: %s, error: %v, skip scanning.", r.Name, err)
-				continue
-			}
-			for _, t := range tags {
-				if err = TriggerImageScan(r.Name, t); err != nil {
-					log.Errorf("Failed to scan image with repository: %s, tag: %s, error: %v.", r.Name, t, err)
-				} else {
-					log.Debugf("Triggered scan for image with repository: %s, tag: %s", r.Name, t)
-				}
+	go scanRepos(repos, regURL)
+	return nil
+}
+
+// ScanImagesByProjectName scans all images under a projet, the whole process will move on if failed to submit any job of a single image.
+func ScanImagesByProjectName(project string) error {
+	regURL, err := config.RegistryURL()
+	if err != nil {
+		log.Errorf("Failed to load registry url")
+		return err
+	}
+	repos, err := dao.GetRepositoryByProjectName(project)
```

Comment:
The method can not be called under integration mode, getting repositories by project ID instead.